### PR TITLE
chore: Prepare 43.0.0 release

### PR DESCRIPTION
**Pull request by andygrove**
## Which issue does this PR close?

Closes #.

### CODE REVIEW

**Review by findepi — APPROVED** [review]

File: dev/release/create-tarball.sh [hunk 1]
```diff
@@ -53,11 +53,6 @@ if [ "$#" -ne 2 ]; then
     exit
 fi
 
-if [[ -z "${GH_TOKEN}" ]]; then
-    echo "Please set personal github token through GH_TOKEN environment variable"
-    exit
-fi
-
```
Comment (andygrove) [inline dev/release/create-tarball.sh:60]: This script does not use GitHub

File: docs/source/user-guide/configs.md [hunk 2]
```diff
@@ -67,7 +67,7 @@ Environment variables are read during `SessionConfig` initialisation so they mus
...
+| datafusion.execution.parquet.created_by                                 | datafusion version 43.0.0 | (writing) Sets "created by" property                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
```
Comment (comphead) [inline docs/source/user-guide/configs.md:70]: I think we can rephrase it to avoid the exact version 🤔

Comment (andygrove) [reply docs/source/user-guide/configs.md:70]: The docs are generated directly from code:

```rust
        /// (writing) Sets "created by" property
        pub created_by: String, default = concat!("datafusion version ", env!("CARGO_PKG_VERSION")).into()
```

Comment (andygrove) [reply docs/source/user-guide/configs.md:70]: It is probably possible to add some special case processing to the macros that generate the docs
